src/interceptors/exception.interceptor.ts

```diff
@@ -0,0 +1,35 @@
+import {
+  ArgumentsHost,
+  Catch,
+  ExceptionFilter,
+  HttpException,
+  Logger,
+} from '@nestjs/common';
+import { Response } from 'express';
+import { APIResponse } from 'src/utils/apiResponse.utils';
+
+@Catch(HttpException)
+export class HttpExceptionFilter implements ExceptionFilter {
+  private logger = new Logger();
+
+  catch(exception: HttpException, host: ArgumentsHost) {
+    let err = exception['response'];
+    err = err.response ? err.response : JSON.stringify(err);
+    const context = host.switchToHttp();
+    const response = context.getResponse<Response>();
+    const status = exception.getStatus();
+    this.logger.error(
+      `Message : ${err}`,
+      `${context.getRequest().method} ${context.getRequest().url}`,
+    );
+    const error =
+      typeof exception.getResponse() === 'string'
+        ? exception.getResponse()
+        : exception.getResponse() &&
+            typeof exception.getResponse()['message'] === 'object'
+          ? exception.getResponse()['message'].join(', ')
+          : exception.getResponse()['message'];
+
+    response.status(status).json(APIResponse.create(0, status, error));
+  }
```

Comment:
PullProofAI

The `HttpExceptionFilter` class seems well-structured and handles exceptions appropriately. However, there are a few areas that can be improved for better error handling and readability:

1. Consider adding more specific error handling logic to differentiate between different types of exceptions.
2. Use TypeScript type checking to ensure the correctness of exception handling.
3. Improve error message formatting for better clarity.

```diff
+ // Example improvement for specific error handling
+ if (exception instanceof CustomException) {
+     // Handle custom exception
+ } else {
+     // Default error handling
+ }
```

Overall, the current implementation is solid, but enhancing it with the suggestions above can make the code more robust and maintainable.



<!-- This is an auto-generated comment by PullProofAI -->